src/details/ArborX_DetailsAlgorithms.hpp

```diff
@@ -68,22 +68,134 @@ bool isValid(Sphere const &s)
   return isValid(s.centroid()) && isFinite(s.radius()) && (s.radius() >= 0.);
 }
 
+class DistanceReturnType
+{
+private:
+  double sq = 0.;
+
+public:
+  // FIXME I do not like that I had to provide a default constructor to be
+  // able to use in the PriorityQueue which holds a C array of
+  // pair<DistanceReturnType, Node*>
+  KOKKOS_FUNCTION DistanceReturnType() = default;
+  // NOTE cannot be declared constexpr because abort() isn't
+  KOKKOS_FUNCTION explicit DistanceReturnType(double v)
+      : sq(v)
+  {
+    /*        if ( KokkosHelpers::isNan( sq ) || sq < 0 )
+                Kokkos::abort( "Invalid arguemnt: DistanceReturnType constructor
+       " "requires non-negative floating-point value" );*/
+  }
+  // TODO consider removing implicit conversion to double
+  KOKKOS_INLINE_FUNCTION operator double() const
+  { /*abort();*/
```

Comment:
Try to uncomment this line and run the benchmark to make sure we do not accidentally compute square roots

Reply:
The code runs also with `abort()`.